### DISCUSSION TITLE
(3차스프린트) 반려 식물 삭제기능 지금 도입해도 될까?

### DISCUSSION THREAD

**Discussion by Choi-JJunho** [woowacourse-teams/2023-pium#168] [질문있어요] [answered]
## 논제

인수테스트를 작성하다보니 반려 식물 삭제에 대한 기능이 없다는 것에 약간 의문을 품었습니다.
사용자가 잘못 등록했거나 식물을 키우지 못하게되었을 경우 삭제해야하는 기능이 필요하지 않을까? 라는 생각이 듭니다.

사실 지금 목표한 내용에는 반려 식물 삭제에 대한 내용은 없으나 구현에 큰 소요가 들지 않는다면 당장 구현해도 문제 없다는 생각이 듭니다.
이에 대한 여러분들의 의견이 듣고싶습니다.

## 예상 API 명세

**DELETE `/pet-plants/{id}`**

- Response body
None

- HttpCode
204

### 요약

반려 식물 삭제 기능 지금 구현해도 괜찮지 않을까요?

**Comment by WaiNaat** [comment]
프론트도 기능 구현은 간단할 것 같긴 한데 데모데이가 얼마 남지 않은 시점에서 새로운 기능을 추가하는 게 맞을지는 모르겠습니다

**Reply by Choi-JJunho** [reply]
기능 협의에서 시간소요가 우려된다면 다음 스프린트 목표로 넘기는 것도 좋아보입니다.

**Comment by Kim0914** [comment, marked as answer]
반려 식물을 삭제한다는 것에 대한 명확한 기준이 필요할 것 같아요 !

참새가 말씀해주신 것과 동일하게 단순히 삭제하는 로직은 백엔드도 구현하는데 큰 소요가 들지 않을 것 같습니다.

다만, 저희가 기획 단계에서 이야기했던 부분이 
반려 식물이 시들어서 더 이상 못키우는 상황에서 해당 식물을 삭제하는 것이 아닌, 기록을 다시 보기위해 남겨둔다 라는 의견을 모았던 것 같습니다 !

제가 말씀드리는 부분이 주노가 말씀해주신 `사용자가 잘못 등록했거나 식물을 키우지 못하게되었을 경우`와는 결이 다른 부분인 것 같긴 한데요 ㅎㅎ

1. 반려 식물을 잘못 등록하거나, 키우지 않는 경우 삭제하는 기능과
2. 반려 식물이 죽거나 시들어서 더 이상 키우지 않지만 기록으로 남겨야 하는 경우

두 가지의 기능을 명확히 구분해서 정의한다면 개발 소요를 확실하게 파악할 수 있을 것 같네요 😀

이와는 별개로 반려 식물을 삭제하는 기능은 꼭 필요하다고 생각해요 !

**Reply by Choi-JJunho** [reply]
`삭제 기능`과 `만료 기능`으로 구분할 수 있겠네요 😮
그레이 말을 들어보니 삭제 기능은 `만료 기능`에 대한 추가적인 논의 이후에 추가해도 좋겠다는 생각이 드네요

**Comment by hozzijeong** [comment]
그레이의 의견에 동의합니다!

정말 입력을 잘못 입력해서 삭제하는 것과 식물을 떠나보내고 그 기록을 간직하는 것과는 차이가 있을것 같아요!

잘못 입력해서 삭제하는 경우는 상세보기에서 버튼 하나만 추가해줘도 되긴 합니다!

**Comment by Choi-JJunho** [comment]
4차 스프린트 시작 전에 반려식물 삭제 및 만료기능에 대한 협의를 진행해보죠!

**Reply by WaiNaat** [reply]
좋아요 질문들어오면 잘 말씀드리겠습니다
